pkg/workflow/spec/fnlib/equals.go

```diff
@@ -0,0 +1,58 @@
+package fnlib
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/puppetlabs/nebula-sdk/pkg/workflow/spec/fn"
+)
+
+var (
+	equalsDescriptor = fn.DescriptorFuncs{
+		DescriptionFunc: func() string { return "Checks if the left side equals the right side" },
+		PositionalInvokerFunc: func(args []interface{}) (fn.Invoker, error) {
+			if len(args) != 2 {
+				return nil, &fn.ArityError{Wanted: []int{2}, Variadic: true, Got: len(args)}
+			}
+
+			fn := fn.InvokerFunc(func(ctx context.Context) (m interface{}, err error) {
+				return isEqual(args[0], args[1])
+			})
+
+			return fn, nil
+		},
+	}
+
+	notEqualsDescriptor = fn.DescriptorFuncs{
+		DescriptionFunc: func() string { return "Checks if the left side does not equal the right side" },
+		PositionalInvokerFunc: func(args []interface{}) (fn.Invoker, error) {
+			if len(args) != 2 {
+				return nil, &fn.ArityError{Wanted: []int{2}, Variadic: true, Got: len(args)}
+			}
+
+			fn := fn.InvokerFunc(func(ctx context.Context) (m interface{}, err error) {
+				result, err := isEqual(args[0], args[1])
+
+				return !result, err
+			})
+
+			return fn, nil
+		},
+	}
+)
+
+func isEqual(left, right interface{}) (bool, error) {
```

Comment:
I'd suggest using `reflect.DeepEqual` here, which will allow us to compare other JSON-y types easily. It would also eliminate the need for an uncomparable type error, which won't really make sense to our users who are passing in JSON/YAML data.

Reply:
I forgot this existed. I haven't used it since the reflect agent days :)